pkg/controllerhelpers/core.go

```diff
@@ -56,6 +56,17 @@ func FindStatusConditionsWithSuffix(conditions []metav1.Condition, suffix string
 	return res
 }
 
+const (
+	// https://github.com/kubernetes/apimachinery/blob/f7c43800319c674eecce7c80a6ac7521a9b50aa8/pkg/apis/meta/v1/types.go#L1640
+	maxReasonLength = 1024
+	// https://github.com/kubernetes/apimachinery/blob/f7c43800319c674eecce7c80a6ac7521a9b50aa8/pkg/apis/meta/v1/types.go#L1648
+	maxMessageLength = 32768
+
+	omissionIndicatorFmt = "\n(%d more omitted)"
```

Comment:
wouldn't it be better to use the separator from func parameters instead of the hardcoded one?